packages/charts/react-charting/src/components/DeclarativeChart/DeclarativeChart.tsx

```diff
@@ -30,6 +32,34 @@ export interface DeclarativeChartProps extends React.RefAttributes<HTMLDivElemen
   chartSchema: any;
 }
 
+const useColorMapping = () => {
+  const colorMapping = React.useRef(new Map<string, string>());
+  const colorIndex = React.useRef(0);
+
+  const totalColors = Object.keys(DataVizPalette).length;
+
+  const getColor = (colorString: string): string => {
```

Comment:
get color should be part of adapter. Only the map should be part of DeclarativeChart

Reply:
moved